Audio/SoundStreamInstance.cpp

```diff
@@ -0,0 +1,681 @@
+//--------------------------------------------------------------------------------------
+// File: SoundStreamInstance.cpp
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+//
+// http://go.microsoft.com/fwlink/?LinkId=248929
+// http://go.microsoft.com/fwlink/?LinkID=615561
+//--------------------------------------------------------------------------------------
+
+#include "pch.h"
+#include "DirectXHelpers.h"
+#include "WaveBankReader.h"
+#include "PlatformHelpers.h"
+#include "SoundCommon.h"
+
+using namespace DirectX;
+
+#ifdef __clang__
+#pragma clang diagnostic ignored "-Wcovered-switch-default"
+#endif
+
+#pragma warning(disable : 4061 4062)
+
+//#define VERBOSE_TRACE
+
+#ifdef VERBOSE_TRACE
+#pragma message("NOTE: Verbose tracing enabled")
+#endif
+
+namespace
+{
+    const size_t DVD_SECTOR_SIZE = 2048;
+    const size_t MEMORY_ALLOC_SIZE = 4096;
+    const size_t MAX_BUFFER_COUNT = 3;
+
+    size_t ComputeAsyncPacketSize(_In_ const WAVEFORMATEX* wfx)
+    {
+        if (!wfx)
+            return 0;
+
+        size_t buffer = wfx->nAvgBytesPerSec * 2;
+        buffer = AlignUp(buffer, MEMORY_ALLOC_SIZE);
+        buffer = std::max<size_t>(65536u, buffer);
```

Comment:
Not sure what kind of upper-bound I should put on this if any. In theory: 64 channels (``XAUDIO2_MAX_CHANNELS``) * 32-bits-per-pixel * 200000 (``XAUDIO2_MAX_SAMPLERATE``) * 2 would be something like 48 MB (and I allocate 3 of them). Not realistic but in theory possible.